plugin-server/tests/worker/ingestion/event-pipeline/extractHeatmapDataStep.test.ts

```diff
@@ -209,6 +213,12 @@ describe('extractHeatmapDataStep()', () => {
         `)
     })
 
+    it('drops if the associated team has explicit opt out', async () => {
+        runner.hub.teamManager.fetchTeam = jest.fn(() => Promise.resolve({ heatmaps_opt_in: false }))
+        const response = await extractHeatmapDataStep(runner, event)
+        expect(response).toEqual([event, []])
```

Comment:
let's add more extensive expects to make sure we don't write to kafka + still strip the event data

```suggestion
        const response = await extractHeatmapDataStep(runner, event)
        expect(response[0]).toEqual(event)
        expect(response[0].properties.$heatmap_data).toBeUndefined()
        expect(response[1]).toHaveLength(0)
        expect(runner.hub.kafkaProducer.produce).toBeCalledTimes(0)
```